tree/treeplayer/src/TTreeReader.cxx

```diff
@@ -298,13 +303,31 @@ void TTreeReader::Initialize()
    }
 }
 
+TFriendProxy &TTreeReader::AddFriendProxy(std::size_t friendIdx)
+{
+   try {
+      return *fFriendProxies.at(friendIdx);
+   } catch (const std::out_of_range &) {
+   }
```

Comment:
Why use this `try {} catch {}` pattern rather than checking for availability?

Reply:
Mmm, do you mean I should check `friendIdx >= fFriendProxies.size()`?

Reply:
Yes, as anyway `at` does that same check to decide to throw .... 
```suggestion
   if (friendIdx >= fFriendProxies.size()) {
      fFriendProxies.resize(friendIdx + 1);
      fFriendProxies[friendIdx] = std::make_unique<ROOT::Internal::TFriendProxy>(fDirector.get(), fTree, friendIdx);
   }
   return *fFriendProxies[friendIdx];
}
```
Albeit the code assumes that we have never `friendIdx >= fFriendProxies.size() +1` (in which case the resize creates a hole with a nullptr that we may dereference later.  So we may as well use:

```suggestion
   if (friendIdx >= fFriendProxies.size()) {
      fFriendProxies.push_bacl(std::make_unique<ROOT::Internal::TFriendProxy>(fDirector.get(), fTree, friendIdx));
   }
   return *fFriendProxies[friendIdx];
}
```
but even that 'feels' weird ... did we mean:
```suggestion
   if (friendIdx >= fFriendProxies.size()) {
      fFriendProxies.resize(friendIdx + 1);
   }
   if (nullptr == fFriendProxies[friendIdx]) {
      fFriendProxies[friendIdx] = std::make_unique<ROOT::Internal::TFriendProxy>(fDirector.get(), fTree, friendIdx);
   }
   return *fFriendProxies[friendIdx];
}
```
(in which case the `catch` may has been a misguided attempt to handle both ifs with one statement but the null (unique)ptr de-reference will seg fault rather than throw as far as I know)

Reply:
Ok, thanks for all the input, I will include it in the next commit